pages/api/pets/index.js

```diff
@@ -1,11 +1,20 @@
 import dbConnect from "@/db/connect";
 import Pet from "@/db/models/Pet";
 
+import { authOptions } from "@/pages/api/auth/[...nextauth]";
+import { getServerSession } from "next-auth/next";
+
 export default async function handler(request, response) {
   await dbConnect();
 
   if (request.method === "GET") {
     try {
+      const session = await getServerSession(request, response, authOptions);
+
+      if (!session) {
+        return response.status(401).json({ message: "Unauthorized" });
+      }
```

Comment:
This piece of code should go up a few lines. You want to check for an active session before you check the request methods. Only if you have an active session do you want to allow access to your available request methods. Otherwise you'd have to implement that check within every single condition.